core/trino-main/src/main/java/io/trino/node/NodeManagerModule.java

```diff
@@ -51,6 +54,14 @@ protected void setup(Binder binder)
             binder.bind(InternalNodeManager.class).to(WorkerInternalNodeManager.class).in(Scopes.SINGLETON);
         }
 
-        install(new AirliftNodeInventoryModule(nodeVersion));
+        install(conditionalModule(
```

Comment:
This can use the new `SwitchModule`
```java
install(switchModule(
        NodeInventoryConfig.class,
        NodeInventoryConfig::getType,
        type -> switch (type) {
            case AIRLIFT_DISCOVERY -> new AirliftNodeInventoryModule(nodeVersion);
            case ANNOUNCE -> new AnnounceNodeInventoryModule();
        }));